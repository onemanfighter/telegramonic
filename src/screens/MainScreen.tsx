import { Outlet } from 'react-router-dom';

const MainScreen = () => {
  return (
    <div>
      <h1>Main Screen</h1>
      <Outlet />
    </div>
  );
};

export default MainScreen;
