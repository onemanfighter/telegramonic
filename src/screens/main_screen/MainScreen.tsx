import { Box } from '@chakra-ui/react';
import { Outlet, useLocation } from 'react-router-dom';
import { Navigation } from './navigation';
import { Footer } from './footer';
import { BreadcrumbComponent } from './breadcrumb';

const MainScreen = () => {
  const location = useLocation();

  const isHomeLocation = location.pathname !== '/home';
  return (
    <Box
      flex={1}
      width={'100%'}
      display={'flex'}
      flexDirection={'column'}
      justifyContent={'flex-start'}
      scrollBehavior={'smooth'}
    >
      <Navigation />
      {isHomeLocation && <BreadcrumbComponent />}
      <Box minHeight={'2xl'} bgColor={'grey.200'}>
        <Outlet />
      </Box>
      <Footer />
    </Box>
  );
};

export default MainScreen;
