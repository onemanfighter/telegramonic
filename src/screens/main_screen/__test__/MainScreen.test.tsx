import { render } from '@testing-library/react';
import MainScreen from '../MainScreen';
import { AppRouterProvider } from '@providers';
import { BrowserRouter } from 'react-router-dom';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '/home/example/path',
  }),
}));

describe.skip('MainScreen', () => {
  it('should render correctly', () => {
    const { container } = render(<MainScreen />, {
      wrapper: BrowserRouter,
    });

    expect(container).toMatchSnapshot();
    const breadcrumb = container.querySelector('BreadcrumbComponent');
    expect(breadcrumb).not.toBeNull();
  });
});
