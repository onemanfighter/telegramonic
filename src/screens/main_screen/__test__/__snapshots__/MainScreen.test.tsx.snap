// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen should render correctly 1`] = `
<div>
  <h2>
    Unexpected Application Error!
  </h2>
  <h3
    style="font-style: italic;"
  >
    win.matchMedia is not a function
  </h3>
  <pre
    style="padding: 0.5rem; background-color: rgba(200, 200, 200, 0.5);"
  >
    TypeError: win.matchMedia is not a function
    at /Users/mr.robot/z-stash/telegramonic/node_modules/@chakra-ui/media-query/src/use-media-query.ts:44:22
    at Array.map (&lt;anonymous&gt;)
    at /Users/mr.robot/z-stash/telegramonic/node_modules/@chakra-ui/media-query/src/use-media-query.ts:42:15
    at commitHookEffectListMount (/Users/mr.robot/z-stash/telegramonic/node_modules/react-dom/cjs/react-dom.development.js:23150:26)
    at commitPassiveMountOnFiber (/Users/mr.robot/z-stash/telegramonic/node_modules/react-dom/cjs/react-dom.development.js:24931:11)
    at commitPassiveMountEffects_complete (/Users/mr.robot/z-stash/telegramonic/node_modules/react-dom/cjs/react-dom.development.js:24891:9)
    at commitPassiveMountEffects_begin (/Users/mr.robot/z-stash/telegramonic/node_modules/react-dom/cjs/react-dom.development.js:24878:7)
    at commitPassiveMountEffects (/Users/mr.robot/z-stash/telegramonic/node_modules/react-dom/cjs/react-dom.development.js:24866:3)
    at flushPassiveEffectsImpl (/Users/mr.robot/z-stash/telegramonic/node_modules/react-dom/cjs/react-dom.development.js:27039:3)
    at flushPassiveEffects (/Users/mr.robot/z-stash/telegramonic/node_modules/react-dom/cjs/react-dom.development.js:26984:14)
    at /Users/mr.robot/z-stash/telegramonic/node_modules/react-dom/cjs/react-dom.development.js:26769:9
    at flushActQueue (/Users/mr.robot/z-stash/telegramonic/node_modules/react/cjs/react.development.js:2667:24)
    at act (/Users/mr.robot/z-stash/telegramonic/node_modules/react/cjs/react.development.js:2582:11)
    at /Users/mr.robot/z-stash/telegramonic/node_modules/@testing-library/react/dist/act-compat.js:63:25
    at renderRoot (/Users/mr.robot/z-stash/telegramonic/node_modules/@testing-library/react/dist/pure.js:159:26)
    at render (/Users/mr.robot/z-stash/telegramonic/node_modules/@testing-library/react/dist/pure.js:246:10)
    at Object.&lt;anonymous&gt; (/Users/mr.robot/z-stash/telegramonic/src/screens/main_screen/__test__/MainScreen.test.tsx:14:33)
    at Promise.then.completed (/Users/mr.robot/z-stash/telegramonic/node_modules/jest-circus/build/utils.js:298:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (/Users/mr.robot/z-stash/telegramonic/node_modules/jest-circus/build/utils.js:231:10)
    at _callCircusTest (/Users/mr.robot/z-stash/telegramonic/node_modules/jest-circus/build/run.js:316:40)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at _runTest (/Users/mr.robot/z-stash/telegramonic/node_modules/jest-circus/build/run.js:252:3)
    at _runTestsForDescribeBlock (/Users/mr.robot/z-stash/telegramonic/node_modules/jest-circus/build/run.js:126:9)
    at _runTestsForDescribeBlock (/Users/mr.robot/z-stash/telegramonic/node_modules/jest-circus/build/run.js:121:9)
    at run (/Users/mr.robot/z-stash/telegramonic/node_modules/jest-circus/build/run.js:71:3)
    at runAndTransformResultsToJestFormat (/Users/mr.robot/z-stash/telegramonic/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:122:21)
    at jestAdapter (/Users/mr.robot/z-stash/telegramonic/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:79:19)
    at runTestInternal (/Users/mr.robot/z-stash/telegramonic/node_modules/jest-config/node_modules/jest-runner/build/runTest.js:367:16)
    at runTest (/Users/mr.robot/z-stash/telegramonic/node_modules/jest-config/node_modules/jest-runner/build/runTest.js:444:34)
  </pre>
  <p>
    ðŸ’¿ Hey developer ðŸ‘‹
  </p>
  <p>
    You can provide a way better UX than this when your app throws errors by providing your own 
    <code
      style="padding: 2px 4px; background-color: rgba(200, 200, 200, 0.5);"
    >
      ErrorBoundary
    </code>
     or
     
    <code
      style="padding: 2px 4px; background-color: rgba(200, 200, 200, 0.5);"
    >
      errorElement
    </code>
     prop on your route.
  </p>
</div>
`;
